### ISSUE TITLE
Build failure on Arch

### ISSUE THREAD

**Issue by mbajor** [Beep6581/RawTherapee#3305]
Hi,
I have a problem with building of RT packages for Arch 32bit (gcc-6.1.1).
There is no problem for builds with -msse2 flag.
[Full log](https://build.opensuse.org/package/live_build_log/home:rawtherapee/rawtherapee-nosse/Arch_Extra/i586)

**Comment by heckflosse** [comment]
@mbajor Your link seems to be wrong :(

**Comment by Beep6581** [comment]
https://build.opensuse.org/package/live_build_log/home:rawtherapee/rawtherapee-nosse/Arch_Extra/i586

**Comment by heckflosse** [comment]
That looks like a gcc-6.1.1 bug...

**Comment by Floessie** [comment]
@heckflosse This is actually a C++11 bug, see [here](http://stackoverflow.com/questions/29750946/is-stdabs0u-ill-formed) and the note [here](http://en.cppreference.com/w/cpp/numeric/math/fabs). I ran into exactly the same problem with my projects at work, where `abs()` was silently degrading `double`s to integers when I switched from C++03 to C++11. The solution was to change `abs()` to `std::fabs()`. which is unambiguous. Well, there were only a few places in my code using `abs()`, I guess that's a bit more work for RT. :-/

HTH,
Flössie

**Comment by Floessie** [comment]
@heckflosse Actually, this here could be more profane: As `<cmath>` is included, `abs()` is now in the `std` namespace, and Marcin's GCC version seems serious about it. I guess, this should fix it:

```patch
diff --git a/rtexif/rtexif.h b/rtexif/rtexif.h
index 2d6c800..471f9fb 100644
--- a/rtexif/rtexif.h
+++ b/rtexif/rtexif.h
@@ -558,7 +558,7 @@ protected:
                     lensAperture = exp( log(a1) + (log(a2) - log(a1)) / (log(f2) - log(f1)) * (log(focalLength) - log(f1)) );
                 }
 
-                dif = abs(lensAperture - maxApertureAtFocal);
+                dif = std::abs(lensAperture - maxApertureAtFocal);
             } else {
                 dif = 0;
             }
```

Nevertheless, one should have an eye on `abs()` due to the problems mentioned above.

**Comment by sguyader** [comment]
I had no problem compiling RT today in my up-to-date Manjaro (Arch-based) distro:

> Branch: master
Version: 4.2.864
Changeset: 918179f621cbd5c863825bd6f720ef5a79c89baa
Compiler: cc 6.1.1
Processor: x86_64
System: Linux
Bit depth: 64 bits
Gtkmm: V2.24.4
Build type: release
Build flags:  -std=gnu++11 -march=native -Werror=unused-label -fopenmp -Werror=unknown-pragmas -O3 -DNDEBUG
Link flags:  -march=native
OpenMP support: ON
MMAP support: ON

**Comment by heckflosse** [comment]
@mbajor Marcin, can you try the patch Floessie provided? I can't test that as I'm on Win64 gcc 5.3.1

**Comment by Floessie** [comment]
@mbajor If it still fails, you could change `std::abs()` to `std::fabs()` and try again.

**Comment by mbajor** [comment]
@heckflosse @Floessie The patch fixed compile error in rtexif.h file, but test build is still in progress
https://build.opensuse.org/public/build/home:marcinbajor:branches:home:rawtherapee/Arch_Extra/i586/rawtherapee-nosse/_log

I didn't test it how it affects on other compilers.

**Comment by mbajor** [comment]
@heckflosse @Floessie Fix confirmed

**Comment by heckflosse** [comment]
@mbajor Thanks for reporting the Issue! @Floessie Thanks for providing the fix!
Fixed with 5cd202ddb563b52044a27a9c46adaafacd6e872f
